src/Atmos/Model/precipitation.jl

```diff
@@ -65,6 +68,9 @@ vars_state(::RainModel, ::Prognostic, FT) = @vars(ρq_rai::FT)
 vars_state(::RainModel, ::Gradient, FT) = @vars(q_rai::FT)
 vars_state(::RainModel, ::GradientFlux, FT) = @vars(∇q_rai::SVector{3, FT})
 
+precompute(::RainModel, atmos::AtmosModel, args, ts, ::Source) =
+    (cache = warm_rain_sources(atmos, args, ts),)
```

Comment:
@trontrytel what if we just did this?
```suggestion
precompute(::RainModel, atmos::AtmosModel, args, ts, ::Source) = warm_rain_sources(atmos, args, ts)
```

Then, below, we'd have something like
```julia
source(::WarmRain_1M{Energy}, m, args) = args.precomputed.precipitation.S_ρ_e
```

Reply:
Personally, it is not more readable to me. You have `args` as arguments everywhere and in the way you actually call the precomputed value. I would keep at least that name separate?